### PR TITLE
Born Digital Workflow and Job Processing

### PR DESCRIPTION
**Pull request by tomcrane**
Create [WorkflowMessage](https://github.com/wellcomecollection/iiif-builder/pull/190/files#diff-11bd245455c6d3b4f89168bc444d9778f30cfc9b0208b59cc20e1ea938b1b5fa) class as per https://github.com/wellcomecollection/platform/issues/5494#issuecomment-1266736867
Start adding support to this to dashboard - if triggered from dashboard, [we "receive" a message](https://github.com/wellcomecollection/iiif-builder/pull/190/files#diff-7e9e7fec1ab660f2537b136440369f3e03fa088a1af6996a23a8bea93fad47f4R52) but not from a queue/topic.

Move FormatFileSize(string) to StringUtils

Added DdsIdentifier::PathElementSafe to complement PackageIdentifierPathElementSafe - if you don't know whether it's a b number or a born digital you can't just use PackageIdentifierPathElementSafe because it will truncate b12312312_0001.

Rename code that uses Goobi*** (e.g., GoobiCallController -> WorkflowCallController)
(except where it's specific to Goobi!)

Add metadata to Canvas for file sizes etc, and added some extension methods to help.

(A note for next PR)
There are two kinds of jobs
Workflow jobs are for packages - process the b number, or CALM id. They may involve many manifestations.
DLCS Ingest jobs are at the manifestation level. They only involve one manifestation (one screen in dashboard) at a time. But they may have more than one batch (units of 100).

### CODE REVIEW

File: src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/DdsInstrumentationContext.cs [hunk 1]
```diff
@@ -50,13 +51,14 @@ public int ClearValidBatches(int from, int to)
             return Database.ExecuteSqlRaw(sql);
         }
 
-        public async Task<WorkflowJob> PutJob(string bNumber, bool forceRebuild, bool take, int? workflowOptions,
+        public async Task<WorkflowJob> PutJob(string identifier, bool forceRebuild, bool take, int? workflowOptions,
```
Comment (donaldgray) [inline src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/DdsInstrumentationContext.cs:54]: should `string identifier` be `DdsIdentifier identifier`?

File: src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Workflow/WorkflowCallRepository.cs [hunk 2]
```diff
@@ -38,6 +39,13 @@ public async Task<WorkflowJob> CreateExpeditedWorkflowJob(string id, int? workfl
             return workflowJob;
         }
 
+        public async Task<WorkflowJob> CreateWorkflowJob(WorkflowMessage message, bool expedite = false)
+        {
+            logger.LogInformation("Creating workflow job from message: " + message);
```
Comment (donaldgray) [inline src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Workflow/WorkflowCallRepository.cs:44]: `logger.LogInformation("Creating workflow job from message: {Message}", message);` for structured logging in Kibana.

Also - is it worth dropping to Debug?

**Review by donaldgray — APPROVED** [review]
LGTM, couple of comments but minor

File: src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Workflow/WorkflowCallRepository.cs (hunk 2 above)
Comment (tomcrane) [reply src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Workflow/WorkflowCallRepository.cs:44]: I must get in the habit of structured logging.
I think the workflow job creation is worth keeping at INFO as something you'd want to track back in the logs. I suspect there are other things that should drop to debug though.

File: src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/DdsInstrumentationContext.cs (hunk 1 above)
Comment (tomcrane) [reply src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/DdsInstrumentationContext.cs:54]: Yes... I'll see how far that cascades if I make that change.

**Pull request merged**